10_datastructures_lab/02.py

```diff
@@ -1,5 +1,20 @@
-"""
-Write a program that prints only unique values
-from sys.argv (without program name)
-"""
-
+""" Find anagrams in word list (read from a file) """
+import sys
+from collections import defaultdict
+from collections import Counter
+# Read input args
+anagrams_file_path = sys.argv[1]
+# Read words into a dictionary holding their histogram as key
+dict_anagrams = defaultdict(list)
+with open(anagrams_file_path,'r') as fin:
+    for word in fin: 
+        # Generate a sorted lower case letters histogram
+        word = word.rstrip().lower()
+        sorted_dict_letters = ""
+        dict_letters = Counter(word)
+        for k in sorted(dict_letters): sorted_dict_letters += k + str(dict_letters[k])

```

Comment:
You don't need to both count and sort - sorting the word's letters is enough. Consider "add" and "dad", when sorted both yield "add"